### DISCUSSION TITLE
Upgrade 2.10.0 - community edition

### DISCUSSION THREAD

**Discussion by klasadoroty** [ant-media/Ant-Media-Server#6496] [General]
Hello everyone. 
I upgraded to version 2.10.0 and... I noticed something like this, described below. I also made a new installation of version 2.10.0 - the same thing.
1 - in APP Settings - this appeared:
"rtspPullTransportType": "3",
shouldn't it be: "rtspPullTransportType": "tcp", ??
This causes RTSP streams /stream source/ to fail to run.
Changing to -tcp- and .. works.

2 - creating endpoints, Add RTMP Endpoint - when we add, for example:
rtmp://login:pass@IP server:1935/APP/NAME.stream this does not work, streams do not reach the selected point.
When we add: rtmp://IP server:1935/APP/NAME.stream .. to the selected point - the stream arrives.
Something is not configured correctly.
Have you found this out as well?

**Comment by yashtandon113** [comment]
@klasadoroty 
Thanks for reporting the issue.

1. It is known that users are facing issues when the setting is `"rtspPullTransportType": "3"` which is TCP or UDP and changing it to TCP only, `"rtspPullTransportType": "1"` resolves the issue. We have also created the [Github issue](https://github.com/ant-media/Ant-Media-Server/issues/6476) on our end to set the default value to 1.

2. For the RTMP end point, the correct format is the second one, which is `rtmp://IP-address:1935/XXX/XXX`. I have not tested it before with the username and password at the RTMP end point.

**Reply by klasadoroty** [reply]
Point 1 - ok, I understand.
Point 2 - .. I disagree here.
You must enter your login and password correctly. Then the target server will allow it.
So how to run EndPoint with login and password?
In lower versions it worked without any problems.

**Reply by yashtandon113** [reply]
@klasadoroty 
The format for the RTMP end point is correct when using username and password. But I have not personally tried it with the user and password.

As you mentioned, it was working with an older version and not with v2.10.0, so let us test this on our end and get back to you.

**Reply by yashtandon113** [reply]
@klasadoroty 
Which platform do you use for publishing RTMP with a username and password?

**Reply by klasadoroty** [reply]
it could be anything... today we /had the opportunity/ and checked it by sending it to the WSE
This should work with anyone.

**Comment by klasadoroty** [comment]
Hi,
..you probably forgot about this problem...
Further releases and the problem as it appeared is still there.
No changes.
Let me remind you: endpoint; like .. rtmp://login:pass@serverip..etc. - does not connect.
In versions below 2.10.0 it worked without a problem. Of course, you can deal with it if you have control over these two servers (AMS and this endpoint).
But the fact is... it does not work.

**Reply by yashtandon113** [reply]
@klasadoroty 
Thanks for the reminder.

I tested the RTMP stream with username and password in 2.10 Community Edition, and it works for me. Here is the FFMPEG command and URL that I used.

`ffmpeg -re -i test.mp4 -c copy -f flv 'rtmp://test:test@localhost/LiveApp/test'`

Because there is no option to configure a username and password on the server side for RTMP publishing, the server ignores the username and password in the RTMP endpoint. In this way, it is not secure and functions as a standard RTMP endpoint without a username or password.

If you want to secure the stream, then you can use the tokens. Check out the [stream security](https://antmedia.io/docs/category/stream-security/) document.

In order to use stream security features, you may consider using the Enterprise Edition of Ant Media Server. If you are interested, you can get the free trial from [here](https://antmedia.io/freetrial/).

I hope this addresses your question.

**Reply by klasadoroty** [reply]
Hi.
You answered off-topic. I'll try again: 'Add RTMP Endpoint'
AMS - standard RTMP endpoint without username and password - correct.
I wrote about the problem with 'endpoint'. When I want to send a working stream from AMS to another server (not AMS) where login/pass is required .. it does not work in version 2.10. In version 2.7 creating endpoint - endpoint; like .. rtmp://login:pass@serverip..etc. - the stream reached another server - note - not AMS(!) and worked. Now it is possible but only without entering login and password.

**Reply by yashtandon113** [reply]
@klasadoroty 

I see, and I apologize for the confusion; thank you for clarifying again.

May I know the platform on which you are sending the RTMP stream with a username and password?

**Reply by klasadoroty** [reply]
... 16 Jul:
https://github.com/orgs/ant-media/discussions/6496#discussioncomment-10064208

**Reply by yashtandon113** [reply]
@klasadoroty 

Let me find a platform to test it with the username and password. I will get back to you as soon as possible.

**Comment by klasadoroty** [comment]
Hi. 2 months have passed.. and you are still looking...
- Add RTMP Endpoint - attempting to transfer a stream from 2.10 to another location/server fails.
Example: rtmp://login:pass@IPSERVER.... does not work.
in version 2.7.0 it works without a problem.
Question: will it be like this now or is it a bug and will be corrected?
Of course, you can deal with it, but so far it was a simple operation - copy/paste /save
The most important thing is: you wrote that you will check and write to us...
